src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -1466,22 +1467,57 @@ because xbuild doesn't support framework reference assemblies.
     DependsOnTargets="_LinkAssembliesNoShrinkInputs"
     Condition="'$(PublishTrimmed)' != 'true'"
     Inputs="@(ResolvedAssemblies);$(_AndroidBuildPropertiesCache)"
-    Outputs="@(ResolvedAssemblies->'$(MonoAndroidIntermediateAssemblyDir)%(DestinationSubPath)')">
+    Outputs="@(ResolvedAssemblies->'$(MonoAndroidIntermediateAssemblyDir)%(DestinationSubPath)');@(ResolvedAssemblies->'$(MonoAndroidIntermediateAssemblyDir)%(DestinationSubDirectory)%(Filename).jlo.xml')">
   <LinkAssembliesNoShrink
+      AlreadyLinked="false"
+      ApplicationJavaClass="$(AndroidApplicationJavaClass)"
+      CodeGenerationTarget="$(AndroidCodegenTarget)"
+      Debug="$(AndroidIncludeDebugSymbols)"
+      EnableMarshalMethods="$(_AndroidUseMarshalMethods)"
+      ErrorOnCustomJavaObject="$(AndroidErrorOnCustomJavaObject)"
+      PackageNamingPolicy="$(AndroidPackageNamingPolicy)"
       ResolvedAssemblies="@(_AllResolvedAssemblies)"
+      ResolvedUserAssemblies="@(ResolvedUserAssemblies)"
       SourceFiles="@(ResolvedAssemblies)"
       DestinationFiles="@(ResolvedAssemblies->'$(MonoAndroidIntermediateAssemblyDir)%(DestinationSubPath)')"
       TargetName="$(TargetName)"
       AddKeepAlives="$(AndroidAddKeepAlives)"
       UseDesignerAssembly="$(AndroidUseDesignerAssembly)"
       Deterministic="$(Deterministic)"
-      ReadSymbols="$(_AndroidLinkAssembliesReadSymbols)"
-  />
+      ReadSymbols="$(_AndroidLinkAssembliesReadSymbols)">
+  </LinkAssembliesNoShrink>
+  
   <ItemGroup>
     <FileWrites Include="$(MonoAndroidIntermediateAssemblyDir)**" />
   </ItemGroup>
 </Target>
 
+<!-- Runs additional steps after ILLink runs (_LinkAssemblies) -->
+<Target Name="_LinkAssembliesAdditionalSteps"
+    DependsOnTargets="_LinkAssembliesNoShrinkInputs"
+    Condition="'$(AndroidLinkMode)' != 'None' OR '$(PublishTrimmed)' == 'true'"
+    Inputs="@(ResolvedAssemblies);$(_AndroidBuildPropertiesCache)"
+    Outputs="@(ResolvedAssemblies->'%(RelativeDir)%(Filename).jlo.xml')">
+  <LinkAssembliesNoShrink
+      AlreadyLinked="true"
```

Comment:
Would it be possible to make a new task:
```csharp
class LinkAssembliesNoShrink : FindJavaObjectsTask // Or some better name
```
Just to reduce confusion around `<LinkAssembliesNoShrink/>` and `AlreadyLinked=true/false`.

The `base` type would write the XML files, and `LinkAssembliesNoShrink` would do the trimmer steps.

Reply:
Yep, I agree that this can be confusing. This seems like a good solution to that confusion.

Reply:
I guess one consideration is whether we think running these additional steps after ILLink will be temporary or not.  If we believe the final goal is to implement them as proper linker steps that ILLink runs then this confusion goes away and we would want to keep them all in `<LinkAssembliesNoShrink>`.

Reply:
I don't know if it is temporary or not. We have some conflicting advice on trimmer steps:

* Don't use trimmer steps.
* Only way to do XYZ: use trimmer steps.

Reply:
Created a new `<AssemblyModifierPipeline>` task.  Although it doesn't actually modify assemblies _today_, it will in the future with the marshal method rewriter step and the remove `[Register]` step.  😉